x-pack/plugins/fleet/public/applications/fleet/sections/settings/components/edit_output_flyout/advanced_options_section.tsx

```diff
@@ -0,0 +1,303 @@
+                  id="xpack.fleet.settings.editOutputFlyout.diskQueueSwitchDescription"
+                  defaultMessage="Once enabled, events will be queued on disk when there's a requirement to cache them on the agents in the event of connection loss."
+                />
+              </EuiText>
+            </EuiFlexItem>
+          </EuiFlexGroup>
+        </EuiFormRow>
+        <EuiSpacer size="m" />
+
+        <EuiFormRow fullWidth {...diskQueueEncryptionEnabled.formRowProps}>
+          <EuiFlexGroup alignItems="flexStart">
+            <EuiFlexItem>
+              <EuiSwitch
+                data-test-subj="editOutputFlyout.diskQueueEncryption"
+                {...diskQueueEncryptionEnabled.props}
+                label={
+                  <FormattedMessage
+                    id="xpack.fleet.settings.editOutputFlyout.diskQueueEncryptionLabel"
+                    defaultMessage="Encryption"
+                  />
+                }
+              />
+            </EuiFlexItem>
+            <EuiFlexItem>
+              <EuiText>
+                <FormattedMessage
+                  id="xpack.fleet.settings.editOutputFlyout.diskQueueEncryptionDescription"
+                  defaultMessage="Enable encryption of data at rest."
+                />
+              </EuiText>
+            </EuiFlexItem>
+          </EuiFlexGroup>
+        </EuiFormRow>
+        <EuiSpacer size="m" />
+        <EuiFormRow
+          fullWidth
+          label={
+            <FormattedMessage
+              id="xpack.fleet.settings.editOutputFlyout.diskQueuePathLabel"
+              defaultMessage="Path"
+            />
+          }
+          {...diskQueuePathInput.formRowProps}
+        >
+          <EuiFieldText
+            fullWidth
+            data-test-subj="settingsOutputsFlyout.diskQueuePath"
+            {...diskQueuePathInput.props}
+            placeholder={i18n.translate(
+              'xpack.fleet.settings.editOutputFlyout.diskQueuePathPlaceholder',
+              {
+                defaultMessage: 'path_data/diskqueue',
+              }
+            )}
+          />
+        </EuiFormRow>
+
+        <EuiFormRow
+          fullWidth
+          {...diskQueueMaxSizeInput.formRowProps}
+          label={
+            <FormattedMessage
+              id="xpack.fleet.settings.editOutputFlyout.diskQueueMaxSize"
+              defaultMessage="Maximum Disk Size"
+            />
+          }
+        >
+          <EuiFlexGroup alignItems="flexStart">
+            <EuiFlexItem>
+              <EuiFieldNumber {...diskQueueMaxSizeInput.props} placeholder="Bytes" min={0} />
+            </EuiFlexItem>
+            <EuiFlexItem>
+              <EuiText>
+                <FormattedMessage
+                  id="xpack.fleet.settings.editOutputFlyout.diskQueueMaxSizeDescription"
+                  defaultMessage="Limits the disk size for spooling of data. If set too low, data may be lost when agent can't write data to the destination."
```

Comment:
I think this description is confusing because "when agent can't write data to the destination" may happen for a number of reasons: no connection, output can't handle throughput, batch size is too small, etc, but this description only indicates no connection.

```suggestion
                  defaultMessage="Limits the disk queue size for spooling of data. When data in the queue exceeds this limit, new events will be dropped."
```